src/Shared/Data/EndCodeLkpRepository.cs

```diff
@@ -0,0 +1,19 @@
+using Microsoft.EntityFrameworkCore;
+
+namespace NHS.ServiceInsights.Data;
+
+public class EndCodeLkpRepository : IEndCodeLkpRepository
+{
+    private readonly ServiceInsightsDbContext _dbContext;
+
+    public EndCodeLkpRepository(ServiceInsightsDbContext dbContext)
+    {
+        _dbContext = dbContext;
+    }
+
+    public async Task<long> GetEndCodeIdAsync(string endCode)
+    {
+        var endCodeLkp = await _dbContext.EndCodeLkps.FirstOrDefaultAsync(ec => ec.EndCode == endCode);
+        return endCodeLkp?.EndCodeId ?? 0;
```

Comment:
Rather than returning 0 we should return null if we don't find an end code. You'll need to change the return type to Task<long?>